apps/common/utils/format.tsx

```diff
@@ -0,0 +1,228 @@
+	** - If smbol is percent, then we will display as `12 %` or `12%` (US)
+	** - If symbol is any other token, we will display as `123,79 USDC` or `USDC 123.79` (US)
+	**********************************************************************************************/
+	let locale = 'fr-FR';
+	if (typeof(navigator) !== 'undefined') {
+		locale = navigator.language || 'fr-FR';
+	}
+	const {shouldDisplaySymbol, shouldCompactValue, ...rest} = options;
+	const intlOptions: Intl.NumberFormatOptions = rest;
+	let isPercent = false;
+	if (symbol && symbol !== '' && shouldDisplaySymbol) {
+		const uppercaseSymbol = String(symbol).toLocaleUpperCase();
+		const symbolToFormat = uppercaseSymbol === 'USD' ? 'USD' : 'EUR';
+		intlOptions.style = uppercaseSymbol === 'PERCENT' ? 'percent' : 'currency',
+		intlOptions.currency = symbolToFormat,
+		intlOptions.currencyDisplay = symbolToFormat === 'EUR' ? 'code' : 'narrowSymbol';
+		isPercent = uppercaseSymbol === 'PERCENT';
+	}
+
+	if (isPercent && amount > 5 && shouldCompactValue) {
+		return (
+			`> ${new Intl.NumberFormat([locale, 'en-US'], intlOptions)
+				.format(5)
+				.replace('EUR', symbol)}`
+		);
+	}
+
+	/* 🔵 - Yearn Finance *************************************************************************
+	** If the amount is above 10k, we will format it to short notation:
+	** - 123947 would be `123,95 k` or  `123.95K` (US)
+	** - 267839372 would be `267,84 M` or  `267.84M` (US)
+	**********************************************************************************************/
+	if (amount > 10_000 && shouldCompactValue) {
+		return (
+			new Intl.NumberFormat([locale, 'en-US'], {
+				...intlOptions,
+				notation: 'compact',
+				compactDisplay: 'short'
+			}).format(amount).replace('EUR', symbol)
+		);
+	}
+
+	/* 🔵 - Yearn Finance *************************************************************************
+	** If the amount is very small, we adjust the decimals to try to display something, up to
+	** "decimals" number of decimals
+	**********************************************************************************************/
+	if (amount < 0.01) {
+		if (amount > 0.00000001) {
+			return (new Intl.NumberFormat([locale, 'en-US'], {
+				...intlOptions,
+				maximumFractionDigits: Math.max(8, intlOptions.maximumFractionDigits || 8)
+			}).format(amount).replace('EUR', symbol));
+		}
+		if (amount > 0.000000000001) {
+			return (new Intl.NumberFormat([locale, 'en-US'], {
+				...intlOptions,
+				maximumFractionDigits: Math.max(12, intlOptions.maximumFractionDigits || 12)
+			}).format(amount).replace('EUR', symbol));
+		}
+		return (new Intl.NumberFormat([locale, 'en-US'], {
+			...intlOptions,
+			maximumFractionDigits: Math.max(decimals, intlOptions.maximumFractionDigits || decimals)
+		}).format(amount).replace('EUR', symbol));
+	}
+	return (
+		new Intl.NumberFormat([locale, 'en-US'], intlOptions)
+			.format(amount)
+			.replace('EUR', symbol)
+	);
+}
```

Comment:
nit: DRY this a bit?

```suggestion
			return formatCurrencyWithPrecision({amount, maxFractionDigits: 8, intlOptions, locale, symbol});
		}
		if (amount > 0.000000000001) {
			return formatCurrencyWithPrecision({amount, maxFractionDigits: 12, intlOptions, locale, symbol});
		}
		return formatCurrencyWithPrecision({amount, maxFractionDigits: decimals, intlOptions, locale, symbol});
	}
	return (
		new Intl.NumberFormat([locale, 'en-US'], intlOptions)
			.format(amount)
			.replace('EUR', symbol)
	);
}

type TFormatCurrencyWithPrecision = {
	amount: number;
	maxFractionDigits: number;
	intlOptions: Intl.NumberFormatOptions;
	locale: string;
	symbol: string;
}

function formatCurrencyWithPrecision({amount, maxFractionDigits, intlOptions, locale, symbol}: TFormatCurrencyWithPrecision): string {
	return new Intl.NumberFormat([locale, 'en-US'], {
		...intlOptions,
		maximumFractionDigits: Math.max(maxFractionDigits, intlOptions.maximumFractionDigits || maxFractionDigits)
	}).format(amount).replace('EUR', symbol);
}
```